tests/van/test_modeling_van.py

```diff
@@ -0,0 +1,273 @@
+    from PIL import Image
+
+    from transformers import AutoFeatureExtractor
+
+
+class VanModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=2,
+        image_size=224,
+        num_channels=3,
+        hidden_sizes=[16, 32, 64, 128],
+        depths=[1, 1, 1, 1],
+        is_training=True,
+        use_labels=True,
+        num_labels=3,
+        scope=None,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.image_size = image_size
+        self.num_channels = num_channels
+        self.hidden_sizes = hidden_sizes
+        self.depths = depths
+        self.is_training = is_training
+        self.use_labels = use_labels
+        self.num_labels = num_labels
+        self.type_sequence_label_size = num_labels
+
+    def prepare_config_and_inputs(self):
+        pixel_values = floats_tensor([self.batch_size, self.num_channels, self.image_size, self.image_size])
+
+        labels = None
+        if self.use_labels:
+            labels = ids_tensor([self.batch_size], self.num_labels)
+
+        config = self.get_config()
+
+        return config, pixel_values, labels
+
+    def get_config(self):
+        return VanConfig(
+            num_channels=self.num_channels,
+            hidden_sizes=self.hidden_sizes,
+            depths=self.depths,
+            num_labels=self.num_labels,
+            is_decoder=False,
+        )
+
+    def create_and_check_model(self, config, pixel_values, labels):
+        model = VanModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values)
+        # expected last hidden states: B, C, H // 32, W // 32
+        self.parent.assertEqual(
+            result.last_hidden_state.shape,
+            (self.batch_size, self.hidden_sizes[-1], self.image_size // 32, self.image_size // 32),
+        )
+
+    def create_and_check_for_image_classification(self, config, pixel_values, labels):
+        model = VanForImageClassification(config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values, labels=labels)
+        self.parent.assertEqual(result.logits.shape, (self.batch_size, self.num_labels))
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, pixel_values, labels = config_and_inputs
+        inputs_dict = {"pixel_values": pixel_values}
+        return config, inputs_dict
+
+
+@require_torch
+class VanModelTest(ModelTesterMixin, unittest.TestCase):
+    """
+    Here we also overwrite some of the tests of test_modeling_common.py, as Van does not use input_ids, inputs_embeds,
+    attention_mask and seq_length.
+    """
+
+    all_model_classes = (VanModel, VanForImageClassification) if is_torch_available() else ()
+
+    test_pruning = False
+    test_torchscript = False
+    test_resize_embeddings = False
+    test_head_masking = False
+    has_attentions = False
```

Comment:
So this model is called Visual Attention Network but it doesn't have attentions 😂

Reply:
What is the definition of attention we use? If by attention you mean a squared matrix `A` where `A[i][j]` tells you the "importance" of `i` with `j` then this model doesn't have attention.

Reply:
Since van do not compute an attention matrix, we don't return anything. Marking as resolved